src/test/java/com/bunq/sdk/BunqSdkTestBase.java

```diff
@@ -4,51 +4,190 @@
+import com.bunq.sdk.model.generated.endpoint.MonetaryAccountBank;
+import com.bunq.sdk.model.generated.endpoint.RequestInquiry;
+import com.bunq.sdk.model.generated.endpoint.SandboxUser;
+import com.bunq.sdk.model.generated.object.Amount;
+import com.bunq.sdk.model.generated.object.Pointer;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.stream.JsonReader;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
 import org.junit.BeforeClass;
 
-import java.util.Arrays;
-import java.util.List;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.UUID;
 
 /**
  * Base class for the Bunq SDK tests.
  */
 public class BunqSdkTestBase {
 
   /**
-   * Description of the test device for Java SDK.
+   * Error constants.
    */
+  private static final String ERROR_COULD_NOT_GENERATE_NEW_API_KEY = "Encountered error while retrieving new sandbox ApiKey.\nError message %s";
+
   private static final String DEVICE_DESCRIPTION = "Java test device";
+  private static final int HTTP_STATUS_OK = 200;
+  private static final String FIELD_RESPONSE = "Response";
+  private static final String FIELD_API_KEY = "ApiKey";
+  private static final int INDEX_FIRST = 0;
+
+  protected static final String TEST_CONFIG_PATH = "bunq-test.conf";
+  protected static final String CONTENT_TYPE = "image/png";
+  protected static final String ATTACHMENT_DESCRIPTION = "TEST PNG JAVA";
+  protected static final String ATTACHMENT_PATH_IN = "assets/bunq_App_Icon_Round@4x.png";
+  protected static final String SCHEME_HTTPS = "https://";
+  protected static final String URL_PATH_SEPARATOR = "/";
+  protected static final String URL_PATH_SANDBOX_USER = "/sandbox-user";
+  protected static final String CURRENCY_EUR = "EUR";
+  protected static final String ACCOUNT_DESCRIPTION = "test account java";
+  protected static final String SPENDING_MONEY_AMOUNT = "500";
+  protected static final String POINTER_TYPE_EMAIL = "EMAIL";
+  protected static final String SUGAR_DADDY_EMAIL = "sugardaddy@bunq.com";
+  protected static final String SUGAR_DADDY_REQUESTS_DESCRIPTION = "sdk java test, thanks daddy <3";
+  protected static final String EMAIL_BRAVO = "bravo@bunq.com";
+  protected static final String CASH_REGISTER_DESCRIPTION = "java cash register test";
+
 
   /**
    * Individual properties.
    */
-  private static String apiKey = Config.getApiKey();
-  private static String[] permittedIps = Config.getPermittedIps();
-  private static String apiConfigPath = Config.getApiConfigPath();
+  protected static MonetaryAccountBank secondMonetaryAccountBank;
+  protected static CashRegister cashRegister;
 
   @BeforeClass
-  public static void setUp() {
+  public static void setUpBeforeClass() {
     BunqContext.loadApiContext(getApiContext());
+    setSecondMonetaryAccountBank();
+    requestSpendingMoney();
+
+    try {
+      Thread.sleep(500);
```

Comment:
Why this sleep?

Reply:
To ensure the spending money request has been accepted before refreshing. The refresh part will be added with #79